keycache/keycache.go

```diff
@@ -78,76 +88,81 @@ func (usage Usage) matches(user string, labels []string) bool {
 }
 
 func NewCache() Cache {
-	return Cache{make(map[string]ActiveUser)}
+	return Cache{make(map[DelegateIndex]ActiveUser)}
 }
 
 // setUser takes an ActiveUser and adds it to the cache.
-func (cache *Cache) setUser(in ActiveUser, name string) {
-	cache.UserKeys[name] = in
+func (cache *Cache) setUser(in ActiveUser, name, slot string) {
+	d := &DelegateIndex{Name: name, Slot: slot}
```

Comment:
Hi, 
I'm just trying to understand the code and learn more about go; is there a reason why you want a pointer to the DelegateTuple and then pass in the dereferenced tuple in line 93?
Thanks